summarizer_server/server.py

```diff
@@ -40,6 +50,70 @@ def summarize():
     return json.dumps(top_sentences)
 
 
+@app.route("/api/feedback/submit", methods=["POST"])
+def submit_feedback():
+    if not request.is_json:
+        abort(400)
+
+    request_payload = request.get_json()
+    url = request_payload.get("url", "")
+    score = request_payload.get("score", "")
+
+    if url == "" or score == "":
+        abort(400)
+
+    email = request_payload.get("email", "")
+    gaia = request_payload.get("gaia", "")
+
+    # check if user wishes to be anonymous
+    user = userservice.get_anonymous()
```

Comment:
Instead of making the `get_anonymous` call on every request, what if this was placed in the else clause of the if statement `if email != "" and gaia != ""`?

Reply:
Didn't know python variable scopes worked like that lol, nice, done!